### PR TITLE
sink(cdc): handle sink errors more fast and light

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve?

Issue Number: ref #8657 

### What is changed and how it works?

Handle sink errors more fast and light, without rebuilding the whole changefeed.

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
To ensure clarity and accuracy, kindly revise the title of your PR and provide comprehensive information in its description.

File: cdc/owner/changefeed.go [hunk 1]
```diff
@@ -296,11 +303,7 @@ func (c *changefeed) tick(ctx cdcContext.Context, captures map[model.CaptureID]*
 		return errors.Trace(err)
 	default:
 	}
-	// we need to wait ddl ddlSink to be ready before we do the other things
-	// otherwise, we may cause a nil pointer panic when we try to write to the ddl ddlSink.
-	if !c.ddlSink.isInitialized() {
-		return nil
-	}
+
```
Comment (nongfushanquan) [inline cdc/owner/changefeed.go:306]: why remove ddl sink initialization check here

Comment (hicqu) [reply cdc/owner/changefeed.go:306]: The DDL internal sink is initialized lazily now, and it's not required that the sink must be initialized when interfaces are called.

**Review by hi-rustin — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**